rust/worker/src/execution/orchestration/get.rs

```diff
@@ -0,0 +1,327 @@
+        if let (Some(logs), Some(segments)) = (
+            self.fetch_log_output.as_ref(),
+            self.fetch_segment_output.as_ref(),
+        ) {
+            let task = wrap(
+                Box::new(self.filter.clone()),
+                FilterInput {
+                    logs: logs.clone(),
+                    segments: segments.clone(),
+                },
+                ctx.receiver(),
+            );
+            self.send_task(ctx, task).await;
+        }
+    }
+}
+
+#[async_trait]
+impl Component for GetOrchestrator {
+    fn get_name() -> &'static str {
+        "Get Orchestrator"
+    }
+
+    fn queue_size(&self) -> usize {
+        self.queue
+    }
+
+    async fn on_start(&mut self, ctx: &ComponentContext<Self>) {
+        let log_task = wrap(Box::new(self.fetch_log.clone()), (), ctx.receiver());
+        let segment_task = wrap(Box::new(self.fetch_segment.clone()), (), ctx.receiver());
+        let log_task_success = self.send_task(ctx, log_task).await;
+        if log_task_success {
+            self.send_task(ctx, segment_task).await;
+        }
+    }
+}
+
+#[async_trait]
+impl Handler<TaskResult<FetchLogOutput, FetchLogError>> for GetOrchestrator {
+    type Result = ();
+
+    async fn handle(
+        &mut self,
+        message: TaskResult<FetchLogOutput, FetchLogError>,
+        ctx: &ComponentContext<Self>,
+    ) {
+        let output = match self.cleanup_response(ctx, message) {
+            Some(output) => output,
+            None => return,
+        };
+        self.fetch_log_output = Some(output);
+        self.try_start_filter_operator(ctx).await;
+    }
+}
+
+#[async_trait]
+impl Handler<TaskResult<FetchSegmentOutput, FetchSegmentError>> for GetOrchestrator {
+    type Result = ();
+
+    async fn handle(
+        &mut self,
+        message: TaskResult<FetchSegmentOutput, FetchSegmentError>,
+        ctx: &ComponentContext<Self>,
+    ) {
+        let output = match self.cleanup_response(ctx, message) {
+            Some(output) => output,
+            None => return,
+        };
+        self.fetch_segment_output = Some(output);
+        self.try_start_filter_operator(ctx).await;
+    }
+}
+
+#[async_trait]
+impl Handler<TaskResult<FilterOutput, FilterError>> for GetOrchestrator {
+    type Result = ();
+
+    async fn handle(
+        &mut self,
+        message: TaskResult<FilterOutput, FilterError>,
+        ctx: &ComponentContext<Self>,
+    ) {
+        let output = match self.cleanup_response(ctx, message) {
+            Some(output) => output,
+            None => return,
+        };
+        let task = wrap(
+            Box::new(self.limit.clone()),
+            LimitInput {
+                logs: self
+                    .fetch_log_output
+                    .as_ref()
+                    .expect("FetchLogOperator should have finished already")
+                    .clone(),
+                segments: self
```

Comment:
i prefer that inputs only consume the segments they need

Reply:
I've deprecated the usage of entire fetch segment output. Now each operator only takes what it needs.